File: AndroidRCLGroupList/app/src/main/java/com/example/shakil/androidrclgrouplist/AlphabetActivity.java
package com.example.shakil.androidrclgrouplist;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.example.shakil.androidrclgrouplist.Alphabet.AlphabetAdapter;
import com.example.shakil.androidrclgrouplist.Interface.IOnAlphabetClickListener;

public class AlphabetActivity extends AppCompatActivity {

    RecyclerView recycler_alphabet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_alphabet);

        recycler_alphabet = findViewById(R.id.recycler_alphabet);
        AlphabetAdapter alphabetAdapter = new AlphabetAdapter();
        alphabetAdapter.setiOnAlphabetClickListener(new IOnAlphabetClickListener() {
            @Override
            public void onAlphabetClickListener(String alphabet, int position) {
                if (position != -1){
                    Intent returnIntent = new Intent();
                    returnIntent.putExtra("result", alphabet);
                    setResult(Activity.RESULT_OK, returnIntent);
                    finish();
                }
            }
        });

        recycler_alphabet.setLayoutManager(new GridLayoutManager(this, 4));
        recycler_alphabet.setAdapter(alphabetAdapter);
    }
}

File: KotlinRCLGroupName/app/src/main/java/com/example/shakil/kotlinrclgroupname/AlphabetActivity.kt
package com.example.shakil.kotlinrclgroupname

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import com.example.shakil.kotlinrclgroupname.Adapter.AlphabetAdapter
import com.example.shakil.kotlinrclgroupname.Interface.IOnAlphabetItemClickListener
import kotlinx.android.synthetic.main.activity_alphabet.*

class AlphabetActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_alphabet)

        val alphabetAdapter = AlphabetAdapter()

        alphabetAdapter.setAlphabetClick(object : IOnAlphabetItemClickListener {
            override fun onAlphabetItemClick(value: String, position: Int) {
                if (position != -1) {
                    val returnIntent = Intent()
                    returnIntent.putExtra("result", value)
                    setResult(Activity.RESULT_OK, returnIntent)
                    finish()
                }
            }
        })

        recycler_alphabet.layoutManager = GridLayoutManager(this, 4)
        recycler_alphabet.adapter = alphabetAdapter
    }
}
